### PR TITLE
New JNL nodes - MIE integration

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information: Same core as #1588 but with new method for building and distributing the data for vehicle nodes, cargo size and offset etc.
    

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in singleplayer?
2. [x] Have you loaded the mission in LAN host?
3. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes: IFA nodes are not updated per agreement, and would need to be redone before IFA is re-enabled

### CODE REVIEW

**Comment by HakonRydland** [comment]
@SonorousMeerkat pls look at this and have it merged into MIE, nodes will not change from this point forward, core system may yet, but this will have no impact on the MIE and its purpose

**Comment by Bob-Murphy** [comment]
@SonorousMeerkat @HakonRydland 
What is with this one?

**Comment by HakonRydland** [comment]
New JNL system

File: A3-Antistasi/Templates/Vanilla/Vanilla_JNL_Nodes.sqf [hunk 1]
```diff
@@ -1,152 +1,198 @@
...
+    ["B_HMG_01_F" call A3A_fnc_classNameToModel,					            [0, 0, 1.19],		        [0, -1, 0],				    4,      200],		        //Static HMG
+    ["B_Mortar_01_F" call A3A_fnc_classNameToModel,                             [-0.1,-0.5,0.79],           [0, 1, 0],                  2,      4000],              //Mortar
+    ["B_HMG_02_high_F" call A3A_fnc_classNameToModel,                           [0.2, -0.5, 1.7],           [0, 1, 0],                  4,      200],               //M2 High
+    ["B_HMG_02_F" call A3A_fnc_classNameToModel,                                [-0.2, 0, 1.3],             [0, -1, 0],                 4,      200],               //M2
+
+    //medium sized crates
+    ["Box_NATO_AmmoVeh_F" call A3A_fnc_classNameToModel,						[0,0,0.85],			    	[1,0,0],		    		2],		                    //Vehicle ammo create
+    ["Land_PaperBox_01_open_boxes_F" call A3A_fnc_classNameToModel,             [0,0,0.85],	                [1,0,0],			    	2],		                    //Stef test supplybox
+    ["Land_FoodSacks_01_cargo_brown_F" call _classNameToModel,                  [0,0,0.85],	                [1,0,0],			    	2],		                    //New city supplies crate
```
Comment (Lazejun) [inline A3-Antistasi/Templates/Vanilla/Vanilla_JNL_Nodes.sqf:170]: Woopsi here
call A3A_fnc_classNameToModel,

File: A3-Antistasi/Templates/3CB/3CB_JNL_Nodes.sqf [hunk 2]
```diff
@@ -1,121 +1,175 @@
 //JNL mounting nodes for cargo and statics.
+//Each element is: [model name, [nodes]]
+//Nodes are build like this: [Available(internal use,  always 1), Hardpoint location, Seats locked when node is in use]
 jnl_vehicleHardpoints append [
```
Comment (Lazejun) [inline A3-Antistasi/Templates/3CB/3CB_JNL_Nodes.sqf:4]: There was another JNL 👀

**Comment by SonorousMeerkat** [comment]
This will want to be redirected at phase 6.5's PR with any UNSUNG data removed unless you want to leave it for future proofing

**Review by SonorousMeerkat — APPROVED** [review]
Nothing MIE wise looks too wrong on a brief look. Provided the actual Logi stuff is solid I'm happy to merge, that will not happen until 6.5 is approved.

**Comment by Bob-Murphy** [comment]
@SonorousMeerkat could you add that to the MIE project please?

**Comment by CalebSerafin** [comment]
Okay, I'll review once changes for https://github.com/official-antistasi-community/A3-Antistasi/pull/1588 have been finalised and added to unstable.

**Comment by Bob-Murphy** [comment]
@CalebSerafin #1588 has been finalized and merged --> This one here can now be rieved

**Comment by HakonRydland** [comment]
> @CalebSerafin #1588 has been finalized and merged --> This one here can now be rieved

hold that, waiting unstable merge up branch by @SonorousMeerkat, then apply new changes from #1588 to this branch.

**Comment by SonorousMeerkat** [comment]
This need some stairs?

**Comment by SonorousMeerkat** [comment]
Done, up to you now

**Comment by CalebSerafin** [comment]
Oh thought this was pointed at unstable.
@HakonRydland What differences are in here compared to New JNL?

**Comment by HakonRydland** [comment]
> Oh thought this was pointed at unstable.
> @HakonRydland What differences are in here compared to New JNL?
the building of the 4 global data structures newJNL has, with MIE its moved to templates so it only add the active mods data to the structure, then its public variabled.
oh and init is now server only

**Comment by HakonRydland** [comment]
branch got fucked, see #1663 for new PR

**Pull request closed without merging**